File: 3047_Find_the_Largest_Area_of_Square_Inside_Two_Rectangles/solution.c
long long largestSquareArea(int** bottomLeft, int bottomLeftSize, int* bottomLeftColSize, int** topRight, int topRightSize, int* topRightColSize) {
    long long max_area = 0;
    for (int i = 0; i < bottomLeftSize; i++)
        for (int j = i + 1; j < bottomLeftSize; j++) {
            long long l = bottomLeft[i][0] > bottomLeft[j][0] ? bottomLeft[i][0] : bottomLeft[j][0];
            long long r = topRight[i][0] < topRight[j][0] ? topRight[i][0] : topRight[j][0];
            long long b = bottomLeft[i][1] > bottomLeft[j][1] ? bottomLeft[i][1] : bottomLeft[j][1];
            long long t = topRight[i][1] < topRight[j][1] ? topRight[i][1] : topRight[j][1];
            if (l < r && b < t) {
                long long s = (r - l < t - b) ? r - l : t - b;
                max_area = (s * s > max_area) ? s * s : max_area;
            }
        }
    return max_area;
}

File: 3047_Find_the_Largest_Area_of_Square_Inside_Two_Rectangles/solution.py
class Solution:
    def largestSquareArea(self, bottomLeft: List[List[int]], topRight: List[List[int]]) -> int:
        max_area, n = 0, len(bottomLeft)
        for i in range(n):
            for j in range(i + 1, n):
                left, right = max(bottomLeft[i][0], bottomLeft[j][0]), min(topRight[i][0], topRight[j][0])
                bottom,top =  max(bottomLeft[i][1], bottomLeft[j][1]), min(topRight[i][1], topRight[j][1])

                if left < right and bottom < top:
                    side = min(right - left, top - bottom)
                    max_area = max(max_area, side * side)

        return max_area
